Lecture_1/Exercise_5/main.c

```diff
@@ -0,0 +1,43 @@
+/// @file
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <stdint.h>
+
+/// Factorial base
+#define FACTORIAL_BASE 10
+
+/*!
+ * Calculate factorial of \b n in cycle (not recursion).
+ *
+ * @param[in]	n Factorial base in range 0-20.
+ * @return	\retval 0 if error occurs
+ *		\retval n! if all ok
+ */
+ uint64_t factorial(uint8_t n)
+ {
+ 	uint64_t tmp = 1;
+ 	if (n == 0) {
+		goto _exit;	// 0! = 1
```

Comment:
Please, avoid using goto in your code. It is not considered to be a good practice.

No changes here, but for future tasks, please use different flow control operators.